src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
+ * guarantees based on the assumptions below
  *
  *  Key assumptions & guarantees of this planner are:
  *  - Trajectories are time-optimal assuming INFINITE JERK capability of the robot
  *
- *  - No speed on the trajectory is larger than the 'max_allowable_speed' parameter
+ *  - No linear speed on the trajectory is larger than the 'max_allowable_linear_speed'
+ * parameter'
+ *  - No angular speed on the trajectory is larger than the 'max_allowable_angular_speed'
+ * parameter
  *
  *  - No acceleration value between points on the trajectory can be larger than the
- * 'max_allowable_acceleration' input parameter
+ * 'max_allowable_linear_acceleration' or 'max_allowable_angular_acceleration input
+ * parameter
  *
  *  - Assuming the grip-limit of the robot IS THE SAME AS THE MAX ACCELERATION then at no
- * point on the path can the sum of centripetal and acceleration force be greater than
- * 'max_allowable_acceleration'
+ *    point on the path can the sum of centripetal and acceleration force be greater than
+ *    'max_allowable_linear_acceleration'
  *
  *  PositionTrajectory generation is done by assuming constant acceleration capability.
  *      - The generator will assume maximum acceleration for the robot between each
  * segment on the path. If using max acceleration breaches either the speed limit of the
  * robot, or the centripetal acceleration limit defined by the curvature and robot speed -
  * the planner will assume the max acceleration that will remain bellow this limit.
  *
- *     - To ensure the robot is capable of deccelerating to react to any sudden changes in
+ *     - To ensure the robot is capable of decelerating to react to any sudden changes in
  * the path, the trajectory is checked for backwards continuity
  *
  * NOTE: The following function is based on an algorithm described in page 18 of
  * http://www2.informatik.uni-freiburg.de/~lau/students/Sprunk2008.pdf
  *
- * @pre path_parameters.num_segments > 2 and num_segments <=
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
- * @pre path_parameters.max_allowable_acceleration > 0
- * @pre path_parameters.max_allowable_speed > 0
- * @pre path_parameters.init_speed >= 0
- * @pre path_parameters.final_speed >= 0
- * @pre The trajectory.trajectory_elements[] array is pre-allocated to handle up to the
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS limit
- *
- * @param path_parameters [in] The data structure including important path parameters as
- * defined by the FirmwareRobotPathParameters struct
- * @param trajectory [out] The planned trajectory that follows robot dynamics limitations
- * with the appropriate guarantees and assumptions outlines above
- *
- * @return The outcome of the trajectory generation. Returns OK if trajectory is valid.
+ * @pre path_parameters members have been completely initialized and have values
+ *
+ * @param position_trajectory [out] The trajectory data struct to be modified to contain
+ * the position trajectory
```

Comment:
How much of this will be "filled up"? If it's going to use the entire array for each member we should make an explicit note indicating as such

Reply:
Added comment stating that path_parameters.num_elements elements will be used in each member array